src/imu/imu.cpp

```diff
@@ -0,0 +1,242 @@
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(driver2->view(ilim), "Unable to open limits interface");
+
+
+    outputPort.open("/test-imu/out");
+    ROBOTTESTINGFRAMEWORK_TEST_REPORT("Opening port "+outputPort.getName()+"...");
+
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getAxes(&axes), "Cannot get number of controlled axes");
+    std::string axisName;
+    std::vector<std::string> axis;
+    
+    for (int i = 0; i < axes; i++)
+    {
+        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(iaxes->getAxisName(i, axisName), "Cannot get the name of controlled axes");
+        axis.push_back(axisName);
+    }
+    
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(model.loadReducedModelFromFile(modelPath.c_str(), axis), Asserter::format("Cannot load model %s", modelPath.c_str()));
+    kinDynComp.loadRobotModel(model.model());
+
+    iDynTree::Vector3 baseLinkOrientationRad;
+    baseLinkOrientationRad.zero();
+
+    baseLinkOrientation = iDynTree::Rotation::RPY(baseLinkOrientationRad[0], baseLinkOrientationRad[1], baseLinkOrientationRad[2]);
+    I_T_base = iDynTree::Transform(baseLinkOrientation, iDynTree::Position::Zero());
+    iDynTree::Twist baseVelocity = iDynTree::Twist::Zero();
+
+    unsigned int dofs = kinDynComp.model().getNrOfDOFs();
+    s.resize(dofs);
+    ds.resize(dofs);
+
+    positions.resize(axes);
+    velocities.resize(axes);
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getEncoders(positions.data()), "Cannot get joint positions");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getEncoderSpeeds(velocities.data()), "Cannot get joint positions");
+
+    for(auto i = 0; i < axes; i++)
+    {
+        s.setVal(i, iDynTree::deg2rad(positions[i]));
+        ds.setVal(i, iDynTree::deg2rad(velocities[i]));
+    }
+
+    gravity.zero();
+    gravity(2) = -9.81;
+
+    kinDynComp.setRobotState(
+        I_T_base,
+        s,
+        baseVelocity,
+        ds,
+        gravity
+    );
+
+    return true;
+}
+
+void Imu::tearDown() {
+    outputPort.interrupt();
+    outputPort.close();
+
+    if(driver1)
+    {
+        delete driver1;
+        driver1 = 0;
+    }
+
+    if(driver2)
+    {
+        delete driver2;
+        driver2 = 0;
+    }
+}
+
+void Imu::run() {
+    ROBOTTESTINGFRAMEWORK_TEST_REPORT("Starting reading IMU orientation values...");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(iorientation->getOrientationSensorMeasureAsRollPitchYaw(0, rpyValues, timestamp), "Unable to obtain rpy measurements.");
+
+    iDynTree::Rotation I_R_FK = kinDynComp.getWorldTransform(frameName).getRotation(); 
+    I_R_I_IMU = (I_R_FK * ((iDynTree::Rotation::RPY(iDynTree::deg2rad(rpyValues[0]), iDynTree::deg2rad(rpyValues[1]), iDynTree::deg2rad(rpyValues[2]))).inverse())); 
+    
+    double minLim;
+    double maxLim;
+
+    for (int i = kinDynComp.model().getJointIndex("neck_pitch"); i <= kinDynComp.model().getJointIndex("neck_yaw"); i++)
```

Comment:
Also this can be generalize it making it a parameter, ideally the user has to specify the 3 joints that have to be controlled, let's remind it for future versions of the test